### PR TITLE
vote functionality

### PR DESCRIPTION
**Pull request by jacobgarner**

### CODE REVIEW

File: src/components/OpenedArticle.jsx [hunk 1]
```diff
@@ -1,18 +1,40 @@
...
 
-    let {articleId} = useParams();
+  const handleUpVoteClick = () => {
+    handleVoteClick(1);
+  };
+  const handleDownVoteClick = () => {
+    handleVoteClick(-1);
+  };
 
+  console.log(article.votes);
```
Comment (nc-modou) [inline src/components/OpenedArticle.jsx]: Remember to remove when you're done debugging

**Review by nc-modou — COMMENTED** [review]
Votes work great so far and the optimistic rendering is on point - however in the case of a vote not working, and the request sending back an error, the optimistic vote should be reversed and the user should be notified somehow to let them know it's not worked. You'll know a request has failed if the code in your catch block ends up running

File: src/api.js [hunk 2]
```diff
@@ -20,3 +20,15 @@ export const getCommentsByArticleId = (articleId) =>{
         return res.data.comments
     })
 }
+
+export const patchArticleVotes = (articleId, amount) =>{
+    if(navigator.onLine){
+    return newsApi.patch(`/articles/${articleId}`, {inc_votes:amount}).then((res)=>{
+        })
+    .catch((err)=>{
```
Comment (popatre) [inline src/api.js:28]: do you need to catch here? You should catch at the end of your chain. You've caught, just to throw again confusingly.

**Review by popatre — APPROVED** [review]
Looks like its been dealt with, however, please see my comment regarding the catch. 

Also, you've got a specific check for the online thing, but i think your .catch stuff elsewhere would deal with this if you sorted out our catch

**Pull request merged**